allennlp/data/samplers/bucket_batch_sampler.py

```diff
@@ -144,5 +153,22 @@ def _guess_sorting_keys(self, instances: Iterable[Instance], num_instances: int
             )
         self.sorting_keys = [longest_padding_key]
 
+    def _expand_sorting_keys(self, schema: Instance) -> None:
```

Comment:
The fact that we need this to exist on the `Sampler` class means this design needs to be revisited.